config/config.go

```diff
@@ -126,17 +132,20 @@ func (cfg *Config) ValidateBasic() error {
 
 // BaseConfig is to marshal/unmarshal toml global config vars
 type BaseConfig struct {
-	Name               string                      `mapstructure:"name,omitempty"`
-	ID                 string                      `mapstructure:"id,omitempty"`
-	BasePath           string                      `mapstructure:"base-path,omitempty"`
-	ChainSpec          string                      `mapstructure:"chain-spec,omitempty"`
-	LogLevel           string                      `mapstructure:"log-level,omitempty"`
-	PrometheusPort     uint32                      `mapstructure:"prometheus-port,omitempty"`
-	RetainBlocks       uint32                      `mapstructure:"retain-blocks,omitempty"`
-	Pruning            pruner.Mode                 `mapstructure:"pruning,omitempty"`
-	PrometheusExternal bool                        `mapstructure:"prometheus-external,omitempty"`
-	NoTelemetry        bool                        `mapstructure:"no-telemetry"`
-	TelemetryURLs      []genesis.TelemetryEndpoint `mapstructure:"telemetry-urls,omitempty"`
+	Name                string                      `mapstructure:"name,omitempty"`
+	ID                  string                      `mapstructure:"id,omitempty"`
+	BasePath            string                      `mapstructure:"base-path,omitempty"`
+	ChainSpec           string                      `mapstructure:"chain-spec,omitempty"`
+	LogLevel            string                      `mapstructure:"log-level,omitempty"`
+	PrometheusPort      uint32                      `mapstructure:"prometheus-port,omitempty"`
+	RetainBlocks        uint32                      `mapstructure:"retain-blocks,omitempty"`
+	Pruning             pruner.Mode                 `mapstructure:"pruning,omitempty"`
+	PrometheusExternal  bool                        `mapstructure:"prometheus-external,omitempty"`
+	NoTelemetry         bool                        `mapstructure:"no-telemetry"`
+	TelemetryURLs       []genesis.TelemetryEndpoint `mapstructure:"telemetry-urls,omitempty"`
+	Checkpoint          bool                        `mapstructure:"checkpoint,omitempty"`
+	CheckpointPath      string                      `mapstructure:"checkpoin-path,omitempty"`
+	CheckpointFrequency uint32                      `mapstructure:"checkpoin-frequency,omitempty"`
```

Comment:
Since the `Checkpoint` has 3 fields, will it make sense to separate it out from BaseConfig and create a separate `CheckpointConfig`